src/main/java/gregtech/common/metatileentities/storage/CraftingRecipeLogic.java

```diff
@@ -120,6 +129,37 @@ public void fillCraftingGrid(Map<Integer, ItemStack> ingredients) {
         updateCurrentRecipe();
     }
 
+    @Override
+    public @Nullable IRecipeTransferError receiveRecipe(@NotNull IRecipeLayout recipeLayout, boolean maxTransfer,
+                                                        boolean simulate) {
+        if (!recipeLayout.getRecipeCategory().getUid().equals(VanillaRecipeCategoryUid.CRAFTING)) {
+            JustEnoughItemsModule.transferHelper.createInternalError();
+        } else if (simulate) {
+            // TODO: highlight missing items in recipe viewer
```

Comment:
Is this missing items from the players inventory, the crafting station inventory, or both? Because one of the possible error returns you list in `IRecipeTransferReciever` seems to do what this todo mentions. Can we just use that error here?